File: leetcode/1-two-sum/1-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        dict = {}  # dictionary to store and access nums
        for i in range(len(nums)):
            if nums[i] in dict:
                return [dict[nums[i]], i]
            
            else:
                dict[target - nums[i]] = i
        

File: 1-two-sum/1-two-sum.go
func twoSum(nums []int, target int) []int {
    lookup := make(map[int]int)
    
    for i, v := range nums {
        j, ok := lookup[-v]
        lookup[v - target] = i
        if ok {
            return []int{j, i}
        }
    }
    return []int{}
}